crates/autopilot/src/database/onchain_order_events/ethflow_events.rs

```diff
@@ -135,6 +139,28 @@ fn convert_to_quote_id_and_user_valid_to(
     Ok((quote_id, user_valid_to))
 }
 
+/// This function determines the block at which to start indexing eth-flow events.
+/// It returns a block that is guaranteed to be before any event emitted by the eth-flow contract *unless* an (optional)
+/// earliest sync start is specified: this parameter takes priority if later than the computed block.
```

Comment:
>  this parameter takes priority if later than the computed block.

This part seems a bit unnecessary. Why is someone manually specifying the start block if it would still get overridden by the automatic block. If the manual block is specified then we should always use it.

Reply:
I modified the logic so that it always uses `skip_event_sync_start` if available instead.